src/combinatorics/simple_graph/connectivity.lean

```diff
@@ -189,6 +206,47 @@ by cases p; simp
 @[simp] lemma end_mem_support {u v : V} (p : G.walk u v) : v ∈ p.support :=
 by induction p; simp [*]
 
+lemma mem_support_iff {u v w : V} (p : G.walk u v) :
+  w ∈ p.support ↔ w = u ∨ w ∈ p.support.tail :=
+by cases p; simp
+
+@[simp]
+lemma mem_tail_support_append_iff {t u v w : V} (p : G.walk u v) (p' : G.walk v w) :
+  t ∈ (p.append p').support.tail ↔ t ∈ p.support.tail ∨ t ∈ p'.support.tail :=
+by rw [tail_support_append, list.mem_append]
+
+@[simp] lemma end_mem_tail_support_of_ne {u v : V} (h : u ≠ v) (p : G.walk u v) :
+  v ∈ p.support.tail :=
+by { obtain ⟨_, _, _, rfl⟩ := exists_eq_cons_of_ne h p, simp }
+
+@[simp]
+lemma mem_support_append_iff {t u v w : V} (p : G.walk u v) (p' : G.walk v w) :
+  t ∈ (p.append p').support ↔ t ∈ p.support ∨ t ∈ p'.support :=
+begin
+  simp only [mem_support_iff, mem_tail_support_append_iff],
+  by_cases h : t = v; by_cases h' : t = u;
+  { subst_vars,
```

Comment:
Maybe `rcases eq_or_ne t v with rfl | h; rcases eq_or_ne t u with rfl | h';` saves you a `subst_vars`?

Reply:
`obtain rfl | h := eq_or_ne t v; obtain rfl | h' := eq_or_ne t u,`, you mean? 😜

Reply:
I'm not sure it's golfing if it ends up being *more* keystrokes 😄 

120 characters:
```
  obtain rfl | h := eq_or_ne t v; obtain rfl | h' := eq_or_ne t u;
  subst_vars;
  try { have := ne.symm h' };
  simp [*],
```

96 characters:
```
  by_cases h : t = v; by_cases h' : t = u;
  subst_vars;
  try { have := ne.symm h' };
  simp [*],
```